Clarification of NextDNS logging policy (#2434)
privacyguides/privacyguides.org: docs/dns.md

[^2]: Cloudflare collects and stores only the limited DNS query data that is sent to the 1.1.1.1 resolver. The 1.1.1.1 resolver service does not log personal data, and the bulk of the limited non-personally identifiable query data is stored only for 25 hours. [https://developers.cloudflare.com/1.1.1.1/privacy/public-dns-resolver/](https://developers.cloudflare.com/1.1.1.1/privacy/public-dns-resolver/)
[^3]: Control D only logs for Premium resolvers with custom DNS profiles. Free resolvers do not log data. [https://controld.com/privacy](https://controld.com/privacy)
[^4]: Mullvad's DNS service is available to both subscribers and non-subscribers of Mullvad VPN. Their privacy policy explicitly claims they do not log DNS requests in any way. [https://mullvad.net/en/help/no-logging-data-policy/](https://mullvad.net/en/help/no-logging-data-policy/)

> [^5]: When used with a user account, NextDNS will enable insights and logging features by default (as some features require it). You can choose retention time and log storage location for any logs you choose to keep, or disable logs altogether. If used without a user account, no data is logged. [https://nextdns.io/privacy](https://nextdns.io/privacy)

Reviewer: ```suggestion
[^5]: When used with an account, NextDNS will enable insights and logging features by default (as some features require it). You can choose retention time and log storage location for any logs you choose to keep, or disable logs altogether. If used without an account, no data is logged. [https://nextdns.io/privacy](https://nextdns.io/privacy)
```

Reviewer 2: Are you sure NextDNS privacy policy number 3 refer to the usage without an account (as the logging can only be requested on an user account)?

We are not trying to cover something, are we? The service clearly violated its policy, why do we still act like nothing happened? An the last question, if this was not NextDNS, do we still recommend the service/app that violated its policy?

> [^5]: When used with a user account, NextDNS will enable insights and logging features by default (note: this seemingly conflicts with their privacy policy). You can choose retention time and log storage location for any logs you choose to keep, or disable logs altogether. If used without a user account, no data is logged. [https://nextdns.io/privacy](https://nextdns.io/privacy)

Reviewer 3: ```suggestion
[^5]: When used with a user account, NextDNS will enable insights and logging features by default as the feature requires it. You can choose retention time and log storage location for any logs you choose to keep, or disable logs altogether. If used without a user account, no data is logged. [https://nextdns.io/privacy](https://nextdns.io/privacy)
```

Reviewer 3: I'm thinking something like this, that clearly states the "logging/insights" feature require some "logging".

Author: @dngray I'm onboard with your amendment. I'm editing it now.

Author: Okay, the PR has been updated. New language:

> When used with a user account, NextDNS will enable insights and logging features by default (as some features require it). You can choose retention time and log storage location for any logs you choose to keep, or disable logs altogether. If used without a user account, no data is logged. [https://nextdns.io/privacy](https://nextdns.io/privacy)

Reviewer 2: @dngray 

> For example with something like GDPR wouldn't this come under "legal basis" being that the feature needs it to work?

Even if the service **needs** the logging to work, I believe that the privacy policy should state it clearly. But currently, it's stated in the opposite direction. 

Moreover, as a DNS resolver service, usage insight is not a hard requirement, e.g. resolving DNS queries. The logging is optional at best.
